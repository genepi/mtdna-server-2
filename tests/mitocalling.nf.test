nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    
    test("Runs with single BAM file and mutserve") {

        when {
            params {
               project = "test-job"
               files = "$projectDir/tests/data/bam/HG00096.mapped.ILLUMINA.bwa.GBR.low_coverage.20101123.bam"
               mode = "mutserve"
               output = "out"
               output_reports = "reports"
               output_auxiliary = "files"
            }
        }

        then {
            assert workflow.success
            assert snapshot(path("${launchDir}/out/variants.annotated.txt")).match()
        }

    }

    test("Runs with BAM file including different header contigs") {

        when {
            params {
               project = "test-job"
               files = "$projectDir/tests/data/bam/mitohpc/*bam"
               mode = "mutserve"
               output = "out"
               output_reports = "reports"
               output_auxiliary = "files"
            }
        }

        then {
            assert snapshot(path("${launchDir}/out/variants.annotated.txt")).match()
        }

    }

    test("Runs with single BAM file and mutect2") {

        when {
            params {
               project = "test-job"
               files = "$projectDir/tests/data/bam/*.bam"
               mode = "mutect2"
               output = "out"
               output_reports = "reports"
               output_auxiliary = "files"
            }
        }

        then {
            assert workflow.success
            assert snapshot(path("${launchDir}/out/variants.annotated.txt")).match()
        }

    }


    test("Runs with two files and one goes through QC") {

        when {
            params {
               project = "test-job"
               files = "$projectDir/tests/data/input_validation/*.bam"
               mode = "mutect2"
               output = "out"
               output_reports = "reports"
               output_auxiliary = "files"
            }
        }

        then {
            assert workflow.success
            assert snapshot(path("${launchDir}/out/variants.annotated.txt"),workflow.trace).match()
        }

    }

    test("Runs with 6 samples provided as testdata with mutserve") {

        setup {
            DownloadUtil.downloadAndExtractZip("https://mitoverse.i-med.ac.at/downloads/6-samples.zip", "${launchDir}/input")
        }

        when {
            params {
               project = "test-job"
               files = "${launchDir}/input/*.bam"
               mode = "mutserve"
               output = "out"
               output_reports = "reports"
               output_auxiliary = "files"
            }
        }

        then {
            assert workflow.success
            assert snapshot(path("${launchDir}/out/variants.annotated.txt"),workflow.trace).match()
        }

    }

    test("Runs with 6 samples provided as testdata with mutect2") {

        setup {
            DownloadUtil.downloadAndExtractZip("https://mitoverse.i-med.ac.at/downloads/6-samples.zip", "${launchDir}/input")
        }

        when {
            params {
               project = "test-job"
               files = "${launchDir}/input/*.bam"
               mode = "mutect2"
               output = "out"
               output_reports = "reports"
               output_auxiliary = "files"
            }
        }

        then {
            assert workflow.success
            assert snapshot(path("${launchDir}/out/variants.annotated.txt"),workflow.trace).match()
        }

    }

    test("Runs with 6 samples provided as testdata with fusion") {

        setup {
            DownloadUtil.downloadAndExtractZip("https://mitoverse.i-med.ac.at/downloads/6-samples.zip", "${launchDir}/input")
        }

        when {
            params {
               project = "test-job"
               files = "${launchDir}/input/*.bam"
               mode = "fusion"
               output = "out"
               output_reports = "reports"
               output_auxiliary = "files"
            }
        }

        then {
            assert workflow.success
            assert snapshot(path("${launchDir}/out/variants.annotated.txt"),workflow.trace).match()
        }

    }

}