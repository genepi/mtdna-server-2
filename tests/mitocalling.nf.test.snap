{
    "Runs with single BAM file and mutect2": {
        "content": [
            "variants.annotated.txt:md5,7dc0d48ba82f76bca62ed2dddbf7d7bb"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-03-02T14:55:31.73871"
    },
    "Runs with 6 samples provided as testdata with mutect2": {
        "content": [
            "variants.annotated.txt:md5,1e00242adc88c7cb44adea0dd77c325e",
            {
                "tasksFailed": 0,
                "tasksCount": 28,
                "tasksSucceeded": 28
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-03-02T15:00:48.608014"
    },
    "Runs with single BAM file and mutserve": {
        "content": [
            "variants.annotated.txt:md5,77be10aad2e2ba4fc50296b693e1d8e0"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-03-02T14:53:52.048799"
    },
    "Runs with BAM file including different header contigs": {
        "content": [
            "variants.annotated.txt:md5,a5345c7718757ded27246ae342b5f25b"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-03-02T13:10:28.843431233"
    },
    "Runs with two files and one goes through QC": {
        "content": [
            "variants.annotated.txt:md5,fb9b8ca4410e67f47bb2f0e4c348f908",
            {
                "tasksFailed": 0,
                "tasksCount": 12,
                "tasksSucceeded": 12
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-03-02T14:56:19.810049"
    },
    "Runs with 6 samples provided as testdata with fusion": {
        "content": [
            "variants.annotated.txt:md5,3a0f3b225f5618d8a3c0d7013911f876",
            {
                "tasksFailed": 0,
                "tasksCount": 40,
                "tasksSucceeded": 40
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-03-02T15:04:51.140209"
    },
    "Runs with 6 samples provided as testdata with mutserve": {
        "content": [
            "variants.annotated.txt:md5,c2f723faf745820f9767548dc52ee39a",
            {
                "tasksFailed": 0,
                "tasksCount": 30,
                "tasksSucceeded": 30
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-03-02T14:57:46.95114"
    }
}